src/Evolution/Systems/GeneralizedHarmonic/BoundaryConditions/BjorhusImpl.cpp

```diff
@@ -0,0 +1,70 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Evolution/Systems/GeneralizedHarmonic/BoundaryConditions/BjorhusImpl.hpp"
+
+#include <array>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Utilities/ContainerHelpers.hpp"
+#include "Utilities/ErrorHandling/Assert.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/Gsl.hpp"
+
+namespace GeneralizedHarmonic::BoundaryConditions::detail {
+template <size_t VolumeDim, typename DataType>
+void set_dt_v_psi_constraint_preserving(
+    const gsl::not_null<tnsr::aa<DataType, VolumeDim, Frame::Inertial>*>
+        bc_dt_v_psi,
+    const tnsr::I<DataType, VolumeDim, Frame::Inertial>&
+        unit_interface_normal_vector,
+    const tnsr::iaa<DataType, VolumeDim, Frame::Inertial>&
+        three_index_constraint,
+    const tnsr::aa<DataType, VolumeDim, Frame::Inertial>&
+        char_projected_rhs_dt_v_psi,
+    const std::array<DataType, 4>& char_speeds) noexcept {
+  ASSERT(get_size(get<0, 0>(*bc_dt_v_psi)) ==
+             get_size(get<0>(unit_interface_normal_vector)),
+         "Size of input variables and temporary memory do not match: "
```

Comment:
Could this message be clearer? (See my suggestion on #2999) 

Also, same question as for #2999 ... why not destructive resize if sizes do not match, instead of an error?

Reply:
Changed to reallocating new memory instead of throwing an error in case of size mismatch. We dont want to destructive resize here as the pointer passed to these impl functions would typically be coming from a `TempBuffer`, and so it won't point to an owning tensor.